src/finalizer/utils/linea/l1ToL2.ts

```diff
@@ -0,0 +1,193 @@
+import { OnChainMessageStatus } from "@consensys/linea-sdk";
+import { L1MessageServiceContract } from "@consensys/linea-sdk/dist/lib/contracts";
+import { TokensRelayedEvent } from "@across-protocol/contracts-v2/dist/typechain/contracts/chain-adapters/Linea_Adapter";
+import { Wallet, utils, providers } from "ethers";
+import { groupBy } from "lodash";
+
+import { HubPoolClient, SpokePoolClient } from "../../../clients";
+import { CONTRACT_ADDRESSES } from "../../../common";
+import { Signer, winston, convertFromWei, TransactionReceipt, paginatedEventQuery } from "../../../utils";
+import { FinalizerPromise, CrossChainTransfer } from "../../types";
+import { initLineaSdk, makeGetMessagesWithStatusByTxHash, MessageWithStatus, lineaAdapterIface } from "./common";
+
+type ParsedAdapterEvent = {
+  parsedLog: utils.LogDescription;
+  log: providers.Log;
+};
+
+export async function lineaL1ToL2Finalizer(
+  logger: winston.Logger,
+  signer: Signer,
+  hubPoolClient: HubPoolClient,
+  spokePoolClient: SpokePoolClient,
+  latestBlockToFinalize: number
+): Promise<FinalizerPromise> {
+  const [l1ChainId, hubPoolAddress] = [hubPoolClient.chainId, hubPoolClient.hubPool.address];
+  const l2ChainId = l1ChainId === 1 ? 59144 : 59140;
+  const lineaSdk = initLineaSdk(l1ChainId, l2ChainId);
+  const l2Contract = lineaSdk.getL2Contract(CONTRACT_ADDRESSES[l2ChainId]?.lineaMessageService.address);
+  const l1Contract = lineaSdk.getL1Contract(CONTRACT_ADDRESSES[l1ChainId]?.lineaMessageService.address);
+  const getMessagesWithStatusByTxHash = makeGetMessagesWithStatusByTxHash(l1Contract, l2Contract);
+
+  // Get Linea's `MessageSent` events originating from HubPool
+  const messageSentEvents = await paginatedEventQuery(
+    // l1Contract.contract.connect(hubPoolClient.hubPool.provider),
+    l1Contract.contract,
+    l1Contract.contract.filters.MessageSent(hubPoolAddress, null, null, null, null, null, null),
```

Comment:
The remaining nulls shouldn't be needed
```suggestion
    l1Contract.contract.filters.MessageSent(hubPoolAddress),
```